src/init/variables/VariableResolver.ts

```diff
@@ -0,0 +1,8 @@
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+
+/**
+ * Converts a key/value object, extracted from the CLI or passed as a parameter,
+ * into a new key/value object where the keys are variables defined in the Components.js configuration.
+ * The resulting values are the values that should be assigned to those variables.
+ */
+export abstract class VariableResolver extends AsyncHandler<Record<string, unknown>, Record<string, unknown>> {}
```

Comment:
or `SettingsResolver`, where Settings is a collection of variables and their values.

Reply:
> How about we create a type for `Record<string, unknown>` and call it something like `Variables` or `VariableSet`. And then reuse everywhere, for clarity.

It might even be interesting to have both a `CliValues` and a `Variables` type, which both would be the same (`Record<string, unknown>`). Then this class would be defined as `VariableResolver extends AsyncHandler<CliValues, Variables>` which makes even more clear what the intended input and output is.

> And how about we just merge `ComputerResolver` and `VariableTransformer` into a single class named `VariableSetTransformer`, given that it only has one implementation anyway?

I do like having the separation between implementation and abstract class just in case we need a new transformer in the future. We also do this in several other places that currently only have 1 implementation.

Reply:
100% agree